api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -0,0 +1,227 @@
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            //fail silently, expected sometimes
+        } finally{
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch(SQLException e){
+
+                }
+            }
+        }
+    }
+
+    public static ResultSet selectForId(Connection c, String storageKey, int id){
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement("SELECT * FROM " + storageKey + " WHERE "
+                    + DatabaseHelper.ID_COL + " = ?;");
+            preparedStatement.setInt(1, id);
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static ResultSet selectFromTable(Connection c, String storageKey, String[] fields, String[]values, Persistable p){
+        org.commcare.modern.database.TableBuilder mTableBuilder = new org.commcare.modern.database.TableBuilder(storageKey);
+        mTableBuilder.addData(p);
+        Pair<String, String[]> mPair = DatabaseHelper.createWhere(fields, values, p);
+
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement("SELECT * FROM " + storageKey + " WHERE " + mPair.first +";");
+            for(int i=0; i<mPair.second.length; i++){
+                preparedStatement.setString(i+1, mPair.second[i]);
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static int insertToTable(Connection c, String storageKey, Persistable p){
+
+        Pair<String, List<Object>> mPair = DatabaseHelper.getTableInsertData(storageKey, p);
+
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement(mPair.first);
+            for(int i=0; i<mPair.second.size(); i++){
+                Object obj = mPair.second.get(i);
+
+                if(obj instanceof String){
+                    preparedStatement.setString(i + 1, (String) obj);
+                } else if(obj instanceof Blob){
+                    preparedStatement.setBlob(i+1, (Blob) obj);
+                } else if(obj instanceof Integer){
+                    preparedStatement.setInt(i + 1, ((Integer) obj).intValue());
+                } else if(obj instanceof byte[]){
+                    preparedStatement.setBinaryStream(i+1,new ByteArrayInputStream((byte[]) obj), ((byte[]) obj).length);
+                }
+            }
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows == 0) {
+                throw new SQLException("Creating user failed, no rows affected.");
+            }
+
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
```

Comment:
Does this file need to be J2ME compat?
Regardless, Try-with-resource requires Android API 19, so you probably need to explicitly close the result set using a finally block.